tests/modules/usercall.c

```diff
@@ -0,0 +1,137 @@
+#include "redismodule.h"
+#include <stdio.h>
+#include <strings.h>
+
+RedisModuleUser *user = NULL;
+
+int call_without_acl(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc < 3) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    size_t cmd_len;
+    const char * cmd = RedisModule_StringPtrLen(argv[1], &cmd_len);
+
+    printf("cmd = %.*s\n", (int) cmd_len, cmd);
+
+    RedisModuleCallReply *reply = RedisModule_Call(ctx, cmd, "Ev", argv + 2, argc - 2);
+    if (reply == NULL) {
+        RedisModule_ReplyWithError(ctx, "Failed to Execute");
+        return REDISMODULE_OK;
+    }
+
+    RedisModule_ReplyWithCallReply(ctx, reply);
+    RedisModule_FreeCallReply(reply);
+    return REDISMODULE_OK;
+}
+
+int call_with_acl(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc < 3) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    size_t cmd_len;
+    const char *cmd = RedisModule_StringPtrLen(argv[1], &cmd_len);
+    RedisModuleCallReply *reply = RedisModule_CallWithUser(ctx, user, cmd, "Ev", argv + 2, argc - 2);
+    if (reply == NULL) {
+        RedisModule_ReplyWithError(ctx, "Failed to Execute");
+        return REDISMODULE_OK;
+    }
+
+    int type = RedisModule_CallReplyType(reply);
+
+    size_t str_len;
+    const char * str = RedisModule_CallReplyStringPtr(reply, &str_len);
+
+    if (type != REDISMODULE_REPLY_ERROR) {
+        RedisModule_ReplyWithCallReply(ctx, reply);
+    } else {
+        RedisModule_ReplyWithError(ctx, str);
+    }
+    RedisModule_FreeCallReply(reply);
```

Comment:
`reply` has been added to auto memory which will result in double free.
https://github.com/redis/redis/blob/ac1cc5a6e15bd73bbb7ee9eff950850fdc69a31a/src/module.c#L6035

Reply:
is this true?  I thought one an still free things with automemory?  i.e. you can manually free strings even when you create a RedisModuleString with a ctx?

Reply:
Ohh, my bad, missed the ctx in `RedisModule_FreeCallReply`.